### PR TITLE
draft: Bach bq UUID

### PR DESCRIPTION
**Pull request by thijs-obj**

### CODE REVIEW

File: bach/bach/dataframe.py [hunk 1]
```diff
@@ -1851,12 +1851,18 @@ def to_pandas(self, limit: Union[int, slice] = None) -> pandas.DataFrame:
         """
         with self.engine.connect() as conn:
             sql = self.view_sql(limit=limit)
-            dtype = {name: series.dtype_to_pandas for name, series in self.all_series.items()
```
Comment (thijs-obj) [inline bach/bach/dataframe.py:1854]: We didn't use this for any Series type, so just removed it

File: bach/bach/dataframe.py [hunk 2]
```diff
@@ -1851,12 +1851,18 @@ def to_pandas(self, limit: Union[int, slice] = None) -> pandas.DataFrame:
...
             # read_sql_query expects a parameterized query, so we need to escape the parameter characters
             sql = escape_parameter_characters(conn, sql)
-            pandas_df = pandas.read_sql_query(sql, conn).astype(dtype)
+            pandas_df = pandas.read_sql_query(sql, conn)
+
+            # post process any columns if needed. e.g. in BigQuery we represent UUIDs as text, so we convert
+            # the strings that the query gives us into UUID objects
+            db_dialect = DBDialect.from_engine(self.engine)
+            for name, series in self.data.items():
+                result_processor = series.query_result_processor.get(db_dialect)
```
[the remark this answers is not in the document]
Comment (jansentom) [reply bach/bach/dataframe.py:1863]: Could we not let pandas [read_sql_query] dtype param (https://pandas.pydata.org/docs/reference/api/pandas.read_sql_query.html) handle this with the a dialect sensitive version of `Series.dtype_to_pandas`? That way there is a much higher chance  we hit an optimized code path in pandas?

**Review by jansentom — APPROVED** [review]

File: bach/bach/dataframe.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (thijs-obj) [reply bach/bach/dataframe.py:1863]: As discussed out-of-band: this doesn't work for uuids, as pandas doesn't have a `uuid` `dtype`, but rather sees uuids  (and strings) as `object`

**Comment by thijs-obj** [comment]
Closing in favour of new PR https://github.com/objectiv/objectiv-analytics/pull/664

**Pull request closed without merging**